be/src/util/zip_util.cpp

```diff
@@ -0,0 +1,154 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+#include "util/zip_util.h"
+#include "util/file_utils.h"
+#include "gutil/strings/substitute.h"
+#include "gutil/strings/util.h"
+#include "env/env.h"
+#include "util/time.h"
+
+namespace doris {
+
+#define DEFAULT_FILE_NAME_SIZE 256
+#define DEFAULT_UNZIP_BUFFER 1048576 // 1MB
+
+#define BREAK_IF_STATUS_ERROR(stmt) { \
+        st = (stmt);                  \
+        if (UNLIKELY(!st.ok())) {     \
+            break;                    \
+        }                             \
+}
+
+using namespace strings;
+
+Status ZipFile::close() {
+    if (_zip_file == nullptr) {
+        unzClose(_zip_file);
+    }
+
+    return Status::OK();
+}
+
+Status ZipFile::extract(const std::string& target_path, const std::string& dir_name) {
+    // check zip file
+    _zip_file = unzOpen64(_zip_path.c_str());
+    if (_zip_file == nullptr) {
+        return Status::InvalidArgument("open zip file: " + _zip_path + " error");
+    }
+
+    unz_global_info64 global_info;
+    int err = unzGetGlobalInfo64(_zip_file, &global_info);
+
+    if (err != UNZ_OK) {
+        return Status::IOError(strings::Substitute("read zip file info $0 error, code: $1", _zip_path, err));
+    }
+
+    // 0.check target path
+    std::string target = target_path + "/" + dir_name;
+    if (FileUtils::check_exist(target)) {
+        return Status::AlreadyExist("path already exists: " + target);
+    }
+
+    // 1.create temp directory
+    std::string temp = target_path + "/.tmp_" + std::to_string(GetCurrentTimeMicros()) + "_" + dir_name;
```

Comment:
I think we can make this temp_director a private member, and remove this directory when destructing.